plugins/module_utils/consul.py

```diff
@@ -309,7 +331,9 @@ def _request(self, method, url_parts, data=None, params=None):
         if 400 <= status < 600:
             raise RequestError(status, response_data)
 
-        return json.loads(response_data)
+        if len(response_data) > 0:
```

Comment:
Can you link me the docs where the result is empty? I have seen an empty string in the consul API but not no content yet :)

Reply:
I hope so. 
If we want to register/deregister a service or check there is no result. Only an HTTP response with an empty result. 
I first tested it with json.loads, but it throws an exception and decides to check if the response_data has data. 
https://developer.hashicorp.com/consul/api-docs/agent/service#register-service
https://developer.hashicorp.com/consul/api-docs/agent/check#register-check

I will test it again to verify it :)

Reply:
You are right, while the docs document that endpoint as "produces application/json" it returns:
```
< HTTP/1.1 200 OK
< Vary: Accept-Encoding
< X-Consul-Default-Acl-Policy: allow
< Date: Tue, 20 Feb 2024 08:11:16 GMT
< Content-Length: 0
```
So all good with your check! Btw it would probably be more pythonic to just do "if response_data" instead of the explicit length check.